st2common/st2common/models/api/base.py

```diff
@@ -94,32 +101,36 @@ def _from_model(cls, model, mask_secrets=False):
         # Special case for models which utilize JSONDictField - there is no need to escape those
         # fields since it contains a JSON string and not a dictionary which doesn't need to be
         # mongo escaped. Skipping this step here substantially speeds things up for that field.
-
-        # Right now we do this here manually for all those fields types but eventually we should
-        # refactor the code to just call unescape chars on escaped fields - more generic and
-        # faster.
         raw_values = {}
-
         for field_name in cls.skip_unescape_field_names:
             if isinstance(doc.get(field_name, None), bytes):
                 raw_values[field_name] = doc.pop(field_name)
-
-        # TODO (Tomaz): In general we really shouldn't need to call unescape chars on the whole doc,
-        # but just on the EscapedDict and EscapedDynamicField fields - doing it on the whole doc
-        # level is slow and not necessary!
-        doc = util_mongodb.unescape_chars(doc)
-
-        # Now add the JSON string field value which shouldn't be escaped back.
-        # We don't JSON parse the field value here because that happens inside the model specific
-        # "from_model()" method where we also parse and convert all the other field values.
```

Comment:
This note should be available in the convert_raw method (either in the docstring or as a comment).